src/include/sof/audio/mux.h

```diff
@@ -57,7 +57,7 @@ struct mux_stream_data {
 
 	uint8_t reserved1[8 - PLATFORM_MAX_CHANNELS]; // padding for extra channels
 	uint8_t reserved2[3]; // padding to ensure proper alignment of following instances
-};
+} __packed;
```

Comment:
We need ``` __attribute__((packed, aligned(4)));``` as I think we have one compiler/arch that does not like ```__packed```

Reply:
Done. Maybe we want to disable the check?
https://github.com/thesofproject/sof/blob/2fbbdb032c127cb0f90fcfc9b9f0879be0009c95/scripts/checkpatch.pl#L5944-L5949

Reply:
ack, IIRC, its the older XCC that does not like the newer ```__packed```.